analyzers/src/SonarAnalyzer.Common/CodeFixContext/SonarCodeFixContext.cs

```diff
@@ -0,0 +1,48 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Text;
+using RoslynCodeFixContext = Microsoft.CodeAnalysis.CodeFixes.CodeFixContext;
+
+namespace SonarAnalyzer.CodeFixContext
+{
+    public readonly struct SonarCodeFixContext
```

Comment:
This should live with the other analysis contexts we have. And follow the same style

Reply:
By living with them I guess you meant to move it to the AnalysisContext folder. Could you tell me what you mind by following the same style?

Reply:
With having way fewer empty lines I agree. But for the classes, I would object. What is the advantage of having them as classes?

Reply:
There should be one empty line between the field and the block of properties.

Reply:
What's the problem the `struct` solves? 

While it's allocation-free, there will be only fixed amount of instances created and they bring maintenance cost (why is it a struct? can it be a class? do I need to implement the IEquatable thing? why isn't it implemented, ...)

Reply:
I think being allocation-free is a good enough reason to make it a struct. And I am unsure about the maintenance costs you mentioned.
As if it would be a class the same could be asked (why is this a struct if the original context is a struct? Can it be a strut?)
Regarding the IEquatable implementation, that can be marked once as a won't fix and it should not pop up ever again.